sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/hash.scala

```diff
@@ -415,7 +415,7 @@ abstract class HashExpression[E] extends Expression {
 
   protected def genHashString(
       ctx: CodegenContext, stringType: StringType, input: String, result: String): String = {
-    if (stringType.supportsBinaryEquality) {
+    if (stringType.supportsBinaryEquality && !stringType.usesTrimCollation) {
```

Comment:
I'm just curious - do we have tests that fail if you remove this change?

Reply:
if not, let's make sure to add them

Reply:
Yes we do, [here](https://github.com/apache/spark/pull/48386/files#diff-11ada8e3fe4dc4efd5e5ddf617e3ff405415f5a9341eb8f1054e79b4fa00538f:~:text=collation%22)%20%7B-,test(%22Support%20Md5%20hash%20expression%20with%20collation%22),-%7B) for example

Reply:
@jovanpavl-db Is the check `!stringType.usesTrimCollation` redundant, btw? `supportsBinaryEquality` should be false when trim is on, shouldn't it?

Reply:
Correct, it happened because pr afterwards changed how supportsBinaryEquality is implemented. Will have cleanup pr for sure :)